FirebaseMessaging/Apps/Sample/Sample/ContentView.swift

```diff
@@ -85,19 +92,22 @@ struct ContentView: View {
               .fontWeight(.semibold)
           }
         }
-        .buttonStyle(IdentityButtonStyle())
-      }
+        Text("\(log)")
+          .lineLimit(10)
+          .multilineTextAlignment(.leading)
+      }.buttonStyle(IdentityButtonStyle())
     }
   }
 
   func getToken() {
     InstanceID.instanceID().instanceID { result, error in
       guard let result = result, error == nil else {
-        print("Failed getting iid and token: ", error ?? "")
+        self.log = "Failed getting iid and token: \(String(describing: error))"
```

Comment:
A Result type would be nice here!

Reply:
Yeah, the token callback can really use a result type here. I'm not sure if we should do it in instanceID or wait til token APIs are ported into Messaging and do it there since we will deprecate instanceID in favor of installationID.

Reply:
Yep, no reason to add IID APIs at this point.